//Custom Exception
    class Program
    {
        public static void Main(string[] args)
        {
            try
            {
                try
                {
                    throw new MyCustomException("My Custom Exception thrown");
                }
                catch (MyCustomException ex)
                {
                    //Console.WriteLine(ex.Message);
                    throw new Exception("My Custom Inner exception thrown", ex);
                }
            }catch(Exception ex)
            {
                Console.WriteLine(ex.Message);
                if (ex.InnerException != null)
                {
                    Console.WriteLine(ex.InnerException.Message);
                }
            }
            Console.ReadKey();
        }
    }

    [Serializable]
    public class MyCustomException : Exception
    {
        public MyCustomException() : base() { }
        public MyCustomException(string message) : base(message) { }
        public MyCustomException(string message, Exception innerException) : base(message, innerException) { }
        public MyCustomException(SerializationInfo info, StreamingContext context) : base(info, context) { }
    }